### ISSUE TITLE
Wrong surface type field when using raw meteorology file

### ISSUE THREAD

**Issue by yuanjianz** [geoschem/gchp#406]
### Name and Institution (Required)

Name: Yuanjian Zhang
Institution: WashU

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://gchp.readthedocs.io/en/latest/reference/SUPPORT.html)
- [x] [User manuals](https://gchp.readthedocs.io/en/latest/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://gchp.readthedocs.io/en/latest/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/GCHP/issues)

## Issue

I am evaluating dust discrepancies between processed wind and raw mass flux GEOS-IT fullchem benchmark. Difference shown in `BudgetConvection` and further `WetLossConvFrac` indicate something is wrong with raw run.
![image](https://github.com/geoschem/GCHP/assets/84134417/c903f28e-eece-4a87-95e6-2ac96a604ff0)
*left panel is processed wind

Over-water `WetLossConvFrac` in raw mass flux run is equivalent to land, in contrary to processed wind run. 
```fortran
    IF ( .NOT. VUD_DEFINED ) THEN
       IF ( State_Met%IsWater(I,J) ) THEN
          VUD = 5e+0_fp
       ELSE
          VUD = 10e+0_fp
       ENDIF
    ENDIF
```
```fortran
    IF ( Vud > TINY_FP ) THEN
       F = 1.0_fp - EXP( -K * TMP / Vud )
    ELSE
       F = 0.0_fp
    ENDIF
```

[GEOS-Chem WetScav module](https://github.com/geoschem/geos-chem/blob/20e2402baf56c682cc04af74adb139efdb6ca000/GeosCore/wetscav_mod.F90) determines `VUD` based on surface type `IsWater` and further determines `WetLossConvFrac`. The image above apparantly indicates there is something wrong with `IsWater` in raw meteorology runs. 

## Test
I ran several tests using
- raw cube-sphere mass flux
- raw cube-sphere wind
- raw lat-lon wind

All raw meteorolgy input runs show wrong surface type over water comparing to processed meteorology input.
![image](https://github.com/geoschem/GCHP/assets/84134417/e26b2e88-48c0-4745-9645-ca14963f2cee)
*left panel is processed wind

However, all meteorological fields (FRLAND, FRSNO, FRLAKE, FRLANDIC, FRSEAICE, FROCEAN) used to determine `IsWater` have slight changes in diagnositics between processed and raw, except `FRSNO`, which is showing strange biases.
![image](https://github.com/geoschem/GCHP/assets/84134417/3d90271a-697e-444a-9a8b-8c7449e8b974)
*left panel is processed wind

## Temporary fix
To further investigate the issue, I replaced `FRSNO` and `FRSEAICE`, which are the only time-varying inputs among those mentioned above, to processed version. The results show the `IsWater` issue is fixed, but I am still not sure why. 

I am doubting it comes from MAPL ExtData handling since the inputs themself seem fine. It would be great if someone familiar with raw meteorology fields and how MAPL handles them can offer some help.

**Comment by lizziel** [comment]
@yuanjianz, interesting. I will take a look at this.

**Comment by lizziel** [comment]
I believe the primary issue you are running into is a bug in use of State_Met%FRSNO when setting the State_Met logicals for surface type (land, water, ice, or snow). [This line](https://github.com/geoschem/geos-chem/blob/20e2402baf56c682cc04af74adb139efdb6ca000/GeosCore/calc_met_mod.F90#L307) should be this instead:
`FRSNO = State_Met%FRLAND(I,J)*State_Met%FRSNO(I,J)`

Here is **before the fix**:

<img width="996" alt="Screenshot 2024-04-17 at 3 06 00 PM" src="https://github.com/geoschem/GCHP/assets/651947/c1ac25f0-9fbe-495c-959b-8ef774fad083">
<img width="1021" alt="Screenshot 2024-04-17 at 3 06 51 PM" src="https://github.com/geoschem/GCHP/assets/651947/05649bed-231a-4960-93db-b31331a69dae">
<img width="1021" alt="Screenshot 2024-04-17 at 3 07 35 PM" src="https://github.com/geoschem/GCHP/assets/651947/f5803817-9c28-41d4-a02f-f7d8f3b27712">

And here is **after the fix**:

<img width="1037" alt="water_after" src="https://github.com/geoschem/GCHP/assets/651947/cbdf7c5f-2f30-4c39-8582-e924450e692f">
<img width="1024" alt="ice_after" src="https://github.com/geoschem/GCHP/assets/651947/e3dae6cb-fd51-455f-a55b-0976925bc7a0">
<img width="1027" alt="snow_after" src="https://github.com/geoschem/GCHP/assets/651947/d9526df6-2e11-4277-a15e-8d987d57bf20">

The question then is why is this happening for raw files only. I suspect it is because the raw files contain missing entries for FRSNO where there is no land. For processed files we set all values to zero except where FRSNO has values. This may also explain discrepancies in the other fraction fields but I need look into this further to verify.

**Comment by lizziel** [comment]
I confirmed that FRSNO is being set equal to missing values in the model (1e15!). These values are not showing up in plots because they correspond exactly to the missing value metadata (or perhaps fillvalue, not sure which). You can see this in the output field for Met_FRSNO.

Met_FRSNO in processed GMAO file run sample output:
```
  0, 2.065287e-09, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 
    0, 0, 0,
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0.0003413867, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 
    0, 0, 0,
  0, 0, 0, 0, 0, 0, 0, 0.003698541, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 
    0, 0,
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ;
```


Met_FRSNO in raw GMAO file run sample output:
```
  0, 2.139596e-09, 0, 0, 0, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, 
    _, _, _,
  0, 0, _, _, 0, _, _, _, _, 0.05538759, _, _, _, _, _, _, _, _, _, _, _, _, 
    _, _,
  0, 0, _, _, _, _, _, 0.0521307, _, _, _, _, _, _, _, _, _, _, _, _, 0, _, 
    _, _,
  0, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _,
  0, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _,
  0, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _ ;
```

Multiplying values by FRLAND effectively removes the missing values. I will put together a PR with a fix that covers all uses of State_Met%FRSNO.

Note that the regridding is also affected by this. Regridding using processed files incorporates zeros during the regridding. Regridding with missing values skips the missing values completely in MAPL. I believe this is why we see the higher values of Met_FRSNO in the raw file run towards the poles.

**Meteorology input files:**
<img width="992" alt="FRSNOW_raw_vs_processed" src="https://github.com/geoschem/GCHP/assets/651947/38fd53a1-b878-4892-a54a-1478b0d85750">


**State_Met%FRSNO output:**
<img width="1065" alt="FRSNO_met" src="https://github.com/geoschem/GCHP/assets/651947/6a0f3103-b90c-47b8-9b34-0430bb2fe828">

**Comment by lizziel** [comment]
There remain small differences between the raw and processed files for the fraction surface types in the constant file. FRSEAICE (time-varying) is exactly the same between raw and processed. I am not sure where these differences in the constants file are coming from. They are small but should be looked at eventually.

Note that I have used GEOS-FP for all of these analyses.
<img width="1062" alt="FRLANDIC" src="https://github.com/geoschem/GCHP/assets/651947/83f3c670-7982-4419-9585-8615d74a68b4">
<img width="1002" alt="FRLAKE" src="https://github.com/geoschem/GCHP/assets/651947/9efe1d16-8597-4b05-9b88-888c70d78403">
<img width="1027" alt="FROCEAN" src="https://github.com/geoschem/GCHP/assets/651947/1cb8539d-1b7f-4df0-8772-f535316c0521">
<img width="1002" alt="FRLAND" src="https://github.com/geoschem/GCHP/assets/651947/d4077490-033c-4f67-bbb0-af62a1112754">

**Comment by yuanjianz** [comment]
Thanks @lizziel, really appreciate your help. The surface type anomaly could have some impacts on raw file fullchem benchmark including wet scavenging and several chemistry processes. I will rerun the raw mass flux fullchem benchmark again after the fix.

**Comment by lizziel** [comment]
See https://github.com/geoschem/geos-chem/pull/2254 for the fix that will go into 14.4.0. The updates are more extensive than the quick fix I gave earlier in this thread, but both are sufficient to address the issue in full chemistry runs using raw GMAO input meteorology files.

**Comment by lizziel** [comment]
The fix for this issue is now merged into 14.4.0.
